app/components/Dashboard/BestNewMusicTab/BestNewMusicMenu/BestNewMusicCard/index.js

```diff
@@ -24,24 +30,94 @@ function toFavoriteTrack({ artist, title, thumbnail }) {
   };
 }
 
+function getReleaseInfo({ artist, title }) {
+  let masterId;
+  return findAlbum(artist, title)
+    .then(findResponse => findResponse.json())
+    .then(findResults => {
+      if (_.isEmpty(findResults) || _.isEmpty(findResults.results)) {
+        return null;
+      }
+
+      return findResults.results[0].master_id;
+    })
+    .then(id => {
+      if (!id) {
+        throw Error(`No release was found for given search criteria. Artist: ${artist}, ablum: ${title}`);
+      }
+
+      masterId = id;
+      return releaseInfo(masterId, 'master');
+    })
+    .then(releaseResponse => releaseResponse.json())
+    .catch(error => {
+      logger.error(error);
+
+      throw error;
+    });
+}
+
+function showSuccessInfo(action, item, settings, type) {
+  const { artist, title, thumbnail } = item;
+  action(
+    `Favorite ${type} added`,
+    `${artist} - ${title} has been added to favorites.`,
+    <img src={thumbnail} />,
+    settings
+  );
+}
+
 const FavoriteIcon = ({ isFavorite, onClick }) =>
   <Icon
     className={styles.card_favorite}
-    name='star'
-    size='large' onClick={onClick}
-    disabled={!isFavorite}
+    name={isFavorite ? 'star' : 'star outline'}
+    size='large'
+    onClick={onClick}
   />;
 
 const BestNewMusicCard = ({
   actions,
   item,
-  favoriteTrack,
+  favoriteItem,
   onClick,
   settings,
-  withFavoriteButton
+  type
 }) => {
   const { artist, title, thumbnail } = item;
 
+  const handleFavoriteIconClick = useCallback((e) => {
+    e.stopPropagation();
+
+    if (favoriteItem) {
+      const removeAction = type === ItemType.ALBUM
+        ? actions.removeFavoriteAlbum
+        : actions.removeFavoriteTrack;
+
+      removeAction(favoriteItem);
+    } else {
+      if (type === ItemType.TRACK) {
+        actions.addFavoriteTrack(toFavoriteTrack(item));
+        showSuccessInfo(actions.info, item, settings, type);
+      }
+
+      if (type === ItemType.ALBUM) {
+        getReleaseInfo(item)
+          .then(releaseItem => {
+            actions.addFavoriteAlbum(releaseItem);
+            showSuccessInfo(actions.info, item, settings, type);
+          })
+          .catch(() => {
+            actions.info(
```

Comment:
I believe handling failed fetch would be nice.

Reply:
This can still be an action, even if it's not storing anything in the redux store. Same for the piece of logic earlier. Anything that performs http requests should be kept out of presentational components.